src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,158 @@
+
+@Service
+public class LectureUnitProcessingService {
+
+    /**
+     * Split units from given file according to given split information.
+     * @param lectureUnitSplitDTOs The split information
+     * @param file The file (lecture slide) to be split
+     * @return The prepared units to be saved
+     */
+    public List<LectureUnitDTO> splitUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, MultipartFile file) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        List<LectureUnitDTO> units = new ArrayList<>();
+        Splitter pdfSplitter = new Splitter();
+        PDDocument document = PDDocument.load(file.getBytes());
+
+        for (LectureUnitSplitDTO lectureUnit : lectureUnitSplitDTOs) {
+            AttachmentUnit attachmentUnit = new AttachmentUnit();
+            Attachment attachment = new Attachment();
+            PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+            pdfSplitter.setStartPage(lectureUnit.startPage());
+            pdfSplitter.setEndPage(lectureUnit.endPage());
+            pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+            List<PDDocument> documentUnits = pdfSplitter.split(document);
+            pdDocumentInformation.setTitle(lectureUnit.unitName());
+            documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
+            documentUnits.get(0).save(outputStream);
+
+            // setup attachmentUnit and attachment
+            attachmentUnit.setDescription("");
+            attachment.setName(lectureUnit.unitName());
+            attachment.setAttachmentType(AttachmentType.FILE);
+            attachment.setVersion(1);
+            attachment.setReleaseDate(lectureUnit.releaseDate());
+            attachment.setUploadDate(ZonedDateTime.now());
+
+            // prepare file to be set from byte[] to MultipartFile by using CommonsMultipartFile
+            String tempDirectory = System.getProperty("java.io.tmpdir");
+            File tempFile = new File(tempDirectory, lectureUnit.unitName() + ".pdf");
+            Files.write(tempFile.toPath(), outputStream.toByteArray());
+            File outputFile = new File(tempFile.toString());
+            FileItem fileItem = new DiskFileItem("mainUnitFile", Files.probeContentType(outputFile.toPath()), false, outputFile.getName(), (int) outputFile.length(),
+                    outputFile.getParentFile());
+            InputStream input = new FileInputStream(outputFile);
+            OutputStream fileItemOutputStream = fileItem.getOutputStream();
+            IOUtils.copy(input, fileItemOutputStream);
+            MultipartFile multipartFile = new CommonsMultipartFile(fileItem);
+
+            LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
+            units.add(lectureUnitsDTO);
+            tempFile.deleteOnExit();
+        }
+
+        document.close();
```

Comment:
Similar to @b-fein's comment about try-with-resources for the OutputStream, also use it for the document.